### PR TITLE
Run version guard checks on PRs

### PR DESCRIPTION
**Pull request by roaks3**
This adds the GitHub Action that checks version guards in erb files that were changed.

Confirmation of an invalid version guard being found in a previous commit here https://github.com/GoogleCloudPlatform/magic-modules/actions/runs/8525807220/job/23353642608?pr=10347

This PR also includes changes to all existing erbs so that they pass the checker (to avoid people touching files and then needing to fix them). You can see from the generated diffs that there are only whitespace differences.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: .github/workflows/mmv1-check-templates.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  pull_request:
+    paths:
+      - 'mmv1/**/*.erb'
+
+jobs:
+  version-guard-check:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
```
Comment (shuyama1) [inline .github/workflows/mmv1-check-templates.yml:15]: I think we'll want to pin the version by a specific hash. Looks like that's how we set up for other workflows after https://github.com/GoogleCloudPlatform/magic-modules/pull/10239

**Review by shuyama1 — COMMENTED** [review]
LGTM - only one small comment

File: .github/workflows/mmv1-check-templates.yml (hunk 1 above)
Comment (roaks3) [reply .github/workflows/mmv1-check-templates.yml]: Nice catch! My copy-paste missed that change, should be fixed now.

**Comment by shuyama1** [comment]
Hey @roaks3, looks like you'll need to rebase the PR due to the CI issue. Sorry about it!

**Comment by roaks3** [comment]
No worries!

**Review by shuyama1 — APPROVED** [review]

**Pull request merged**